logger.go

```diff
@@ -227,7 +258,15 @@ func (logger *Logger) ListHooks() []string {
 func (logger *Logger) refreshLoggers() *Logger {
 	logger.Logger.Hooks = make(logrus.LevelHooks)
 	var level logrus.Level
-	for _, hook := range logger.hooks {
+
+	names := make([]string, 0, len(logger.hooks))
```

Comment:
No need to init your map's length if you're going to append

Reply:
It is not a map and it is more efficient that way.  That avoid multiple memory allocations.

Reply:
But, I change the code to reuse the ListHooks() function which already sorted the hooks. This is needed to ensure test idempotency.